MacOS Apple Silicon - igraph 0.10.4 - 328 - example::safelocale (Failed)

## Issue description
```
nix --version
nix (Nix) 2.15.0

ProductName: macOS
ProductVersion: 13.3.1
BuildVersion: 22E261

error: builder for '/nix/store/5ylab67jw0m69y09gb5vv1s9xg61h481-igraph-0.10.4.drv' failed with exit code 2;
       > make[2]: *** [CMakeFiles/Makefile2:2709: tests/CMakeFiles/check.dir/all] Error 2
       > make[2]: Leaving directory '/private/tmp/nix-build-igraph-0.10.4.drv-0/source/build'
       > make[1]: *** [CMakeFiles/Makefile2:2716: tests/CMakeFiles/check.dir/rule] Error 2
       > make[1]: Leaving directory '/private/tmp/nix-build-igraph-0.10.4.drv-0/source/build'
       > make: *** [Makefile:810: check] Error 2
       For full logs, run 'nix-store -l /nix/store/5ylab67jw0m69y09gb5vv1s9xg61h481-igraph-0.10.4.drv'.

328/537 Test #328: example::safelocale.................................***Failed    1.68 sec
Error at src/io/parse_utils.c:155 : Unexpected character '.' while parsing real number. - Parse error.
Test exited abnormally with error: Subprocess aborted
=========================================

CMake Error at /tmp/nix-build-igraph-0.10.4.drv-0/source/etc/cmake/run_legacy_test.cmake:32 (message):
  Exiting test.

99% tests passed, 1 tests failed out of 537

Total Test time (real) =  86.32 sec

The following tests FAILED:
        328 - example::safelocale (Failed)
Errors while running CTest
make[3]: *** [tests/CMakeFiles/check.dir/build.make:71: tests/CMakeFiles/check] Error 8
make[3]: Leaving directory '/private/tmp/nix-build-igraph-0.10.4.drv-0/source/build'
make[2]: *** [CMakeFiles/Makefile2:2709: tests/CMakeFiles/check.dir/all] Error 2
make[2]: Leaving directory '/private/tmp/nix-build-igraph-0.10.4.drv-0/source/build'
make[1]: *** [CMakeFiles/Makefile2:2716: tests/CMakeFiles/check.dir/rule] Error 2
make[1]: Leaving directory '/private/tmp/nix-build-igraph-0.10.4.drv-0/source/build'
make: *** [Makefile:810: check] Error 2
```

## Technical details

```
 nix-shell -p nix-info --run "nix-info -m"
these 3 paths will be fetched (1.01 MiB download, 8.01 MiB unpacked):
  /nix/store/jxs932c96jzaa5n88vgp1vzimd3k2dxq-DarwinTools-1
  /nix/store/4zq9fc233hy1c7agkk9ffn2wr2y30yra-bash-interactive-5.2-p15
  /nix/store/v97x2zbzwpxr2djak6pr2zhvdhn2qqi3-nix-info
copying path '/nix/store/jxs932c96jzaa5n88vgp1vzimd3k2dxq-DarwinTools-1' from 'https://cache.nixos.org'...
copying path '/nix/store/4zq9fc233hy1c7agkk9ffn2wr2y30yra-bash-interactive-5.2-p15' from 'https://cache.nixos.org'...
copying path '/nix/store/v97x2zbzwpxr2djak6pr2zhvdhn2qqi3-nix-info' from 'https://cache.nixos.org'...
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.4.0, macOS 13.3.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.15.0`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

```

--- a/pkgs/development/libraries/igraph/default.nix
+++ b/pkgs/development/libraries/igraph/default.nix
@@ -37,6 +37,10 @@ stdenv.mkDerivation (finalAttrs: {
 
   postPatch = ''
     echo "${finalAttrs.version}" > IGRAPH_VERSION
+  ''
+  # https://github.com/igraph/igraph/issues/2340
+  + lib.optionalString stdenv.isDarwin ''
+    sed -i "/safelocale/d" tests/CMakeLists.txt
   '';
 
   outputs = [ "out" "dev" "doc" ];

